### PR TITLE
Read all the workspace after a run and write inputs into it; fixes #5

### LINKED ISSUE #5: DST-MAT-SR-0020 : Read/Write the Matlab Workspace
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the DEHP-MATLAB
- [x] I have searched [open](https://github.com/RHEAGROUP/DEHP-MATLAB/issues) and [closed](https://github.com/RHEAGROUP/DEHP-MATLAB/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

The Matlab DST adapter shall be able to read & write the Matlab workspace.

### System Configuration

- DEHPCommon version:
- Environment (Operating system, version and so on):
- .NET Framework version:
- Additional information:

### PR DESCRIPTION
**Pull request by antoineatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEHP-MATLAB/pulls) open
- [x] I have verified that I am following the DEHP-MATLAB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEHP-MATLAB/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixes #5 
- Read all the Matlab Workspace after a run
- Write all detected inputs into the Workspace
- Write changes into the Workspace while we change an input 
- Added some async Task 
- Connection dialog added

### CODE REVIEW

**Comment by antoineatrhea** [comment]
image

File: DEHPMatlab/DstController/DstController.cs [hunk 1]
```diff
@@ -80,13 +90,24 @@ public class DstController : ReactiveObject, IDstController
...
             this.matlabParser = matlabParser;
+            this.statusBar = statusBar;
             this.InitializeObservables();
         }
 
+        /// <summary>
+        /// Destructor of the <see cref="DstController"/>.
+        /// Handles the destruction of the loaded script.
+        /// </summary>
+        ~DstController()
```
Comment (alexatrhea) [inline DEHPMatlab/DstController/DstController.cs:106]: We never implement destructors. Is there no other way to implement the unloading?

File: DEHPMatlab/DstController/DstController.cs [hunk 2]
```diff
@@ -176,8 +234,9 @@ public void Disconnect()
         public void LoadScript(string scriptPath)
         {
             this.UnloadScript();
+            this.MatlabWorkspaceInputRowViewModels.Clear();
 
-            var detectedInputs = this.matlabParser.ParseMatlabScript(scriptPath, 
+            List<MatlabWorkspaceRowViewModel> detectedInputs = this.matlabParser.ParseMatlabScript(scriptPath,
```
Comment (alexatrhea) [inline DEHPMatlab/DstController/DstController.cs:239]: do implicit `var`

File: DEHPMatlab/DstController/DstController.cs [hunk 3]
```diff
@@ -193,35 +252,74 @@ public void LoadScript(string scriptPath)
...
+        /// <returns>The <see cref="Task"/></returns>
+        public async Task LoadMatlabWorkspace()
         {
             if (this.IsSessionOpen)
             {
-                foreach (var matlabWorkspaceInputRowViewModel in this.MatlabWorkspaceInputRowViewModels)
+                this.MatlabAllWorkspaceRowViewModels.Clear();
+                var uniqueVariable = $"uv{DateTime.Now:yyyyMMddHHmmss}";
+                await this.matlabConnector.ExecuteFunction($"{uniqueVariable} = who");
+                List<MatlabWorkspaceRowViewModel> variables = new List<MatlabWorkspaceRowViewModel>();
```
Comment (alexatrhea) [inline DEHPMatlab/DstController/DstController.cs:306]: do implicit `var` and add some empty lines to make it a bit more readable.

File: DEHPMatlab/Services/MatlabConnector/MatlabConnector.cs [hunk 4]
```diff
@@ -91,65 +92,67 @@ public MatlabConnector(IStatusBarControlViewModel statusBarControl)
         /// Launch a Matlab Instance and connects to it
         /// </summary>
         /// <param name="comInteropName">The name of the COM Interop</param>
-        public void Connect(string comInteropName)
+        /// <returns>The <see cref="Task"/></returns>
+        public async Task Connect(string comInteropName)
```
Comment (alexatrhea) [inline DEHPMatlab/Services/MatlabConnector/MatlabConnector.cs:96]: For all the mothods in this class I am not certain why do you want to do internal `Task.Run()` if they are already awaited upstream? Dont you do then double `await`? As in somewhere you already call `await MatlabConnector.Connect(string)` no?

File: DEHPMatlab/ViewModel/Dialogs/DstConnectViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+        /// <summary>
+        /// Try to connect to a new instance of Matlab
+        /// </summary>
+        /// <returns>The <see cref="Task"/></returns>
+        private async Task ConnectCommandExecute()
+        {
+            this.IsBusy = true;
+
+            var correctVersion = this.SelectedMatlabVersion.Key == "latest" ? BaseMatlabKey : $"{BaseMatlabKey}.{this.SelectedMatlabVersion.Key}";
+            await this.dstController.Connect(correctVersion);
```
Comment (alexatrhea) [inline DEHPMatlab/ViewModel/Dialogs/DstConnectViewModel.cs:116]: Here for example...

File: DEHPMatlab/ViewModel/Dialogs/DstConnectViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+            this.ConnectCommand = ReactiveCommand.CreateAsyncTask(async _ => await this.ConnectCommandExecute());
+        }
+
+        /// <summary>
+        /// Populate the <see cref="MatlabVersionDictionary"/>
+        /// </summary>
+        private void PopulateDictionary()
+        {
+            this.MatlabVersionDictionary["latest"] = "Lastest Installed Version";
+            this.MatlabVersionDictionary["9.11"] = "Matlab R2021b";
```
Comment (alexatrhea) [inline DEHPMatlab/ViewModel/Dialogs/DstConnectViewModel.cs:89]: This can be done a bit smarter in terms of repetition and being prone to errors.

**Review by alexatrhea — CHANGES_REQUESTED** [review]
Some changes requested

**Review by nathanatrhea — CHANGES_REQUESTED** [review]

**Review by alexatrhea — APPROVED** [review]
approved pending Nathans changes

**Review by nathanatrhea — APPROVED** [review]

**Pull request merged**